onboarding/source/amb_light_service.c

```diff
@@ -33,21 +53,83 @@ static void lightServiceTask(void * pvParameters);
 
 obc_error_code_t initLightService(void) {
     /* USER CODE BEGIN */
-    // Create the task and queue here. Return error code if task/queue was not created successfully.
+    if (lightServiceHandle == NULL) {
+        lightServiceHandle = xTaskCreateStatic(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                lightServiceStack,
+                                &lightServiceBuffer);
+                                    }
+    if (lightServiceHandle == NULL)
+    {
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    }
 
-    /* USER CODE END */
+    if (lightserviceQueue == NULL) {
+        lightserviceQueue = xQueueCreateStatic(LIGHT_QUEUE_LENGTH, LIGHT_QUEUE_SIZE,uclightQueueStorageArea, &xlightStaticQueue);
+
+    if (lightserviceQueue == NULL) {
+        return OBC_ERR_CODE_INVALID_ARG;
+    }
+    }
+return OBC_ERR_CODE_SUCCESS;                           
+/* USER CODE END */
+}
+
+uint32_t getLightSensorData(void)
+{
+	adcData_t adcData;
+	
+
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, &adcData);
+	
+	return adcData.value;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    light_event_t eventReceived;
+
+    while (1) {
+        if (xQueueReceive(lightserviceQueue, &eventReceived, portMAX_DELAY) == pdPASS) {
+            
+            if (eventReceived == MEASURE_LIGHT) {
+                
 
+                char str[32];
+                uint16_t lightdata = getLightSensorData();
+                    snprintf(str, 32, "The ambient light is %u", lightdata);
+                    /** - Transmit the Conversion data to PC using SCI*/
+                    sciPrintf("%u\n", str);
```

Comment:
`sciPrintf` basically works the same as `stdio`'s `printf` function. Read more about it here: https://cplusplus.com/reference/cstdio/printf/

No need to use `snprintf` here. Also, no need for the `str` array